src/plugins/auto-closer/index.js

```diff
@@ -0,0 +1,172 @@
+not something the team can commit to.
+
+Thanks for contributing to ESLint and we appreciate your understanding.
+
+[//]: # (auto-close)
+`;
+}
+
+/**
+ * Creates the bot comment to leave on auto-closed unaccepted issues.
+ * @returns {string} comment message
+ * @private
+ */
+function createUnacceptedAutoCloseMessage() {
+    return `
+Unfortunately, it looks like there wasn't enough interest from the team
+or community to implement this change. While we wish we'd be able to
+accommodate everyone's requests, we do need to prioritize. We've found
+that issues failing to reach accepted status after 21 days tend to
+never be accepted, and as such, we [close those issues](https://eslint.org/docs/maintainer-guide/issues#when-to-close-an-issue).
+This doesn't mean the idea isn't interesting or useful, just that it's
+not something the team can commit to.
+
+Thanks for contributing to ESLint and we appreciate your understanding.
+
+[//]: # (auto-close)
+`;
+}
+
+/**
+ * Closes an issue and adds the auto-close label.
+ * @param {probot.Context} context Probot context for the current repository
+ * @param {number} issueNum The issue number on the current repository
+ * @param {string} commentText The text of the comment to post on the issue.
+ * @returns {Promise<void>} A Promise that fulfills when the issue has been archived
+ */
+async function closeIssue(context, issueNum, commentText) {
+    await Promise.all([
+        context.github.issues.edit(context.repo({ number: issueNum, state: "closed" })),
+        context.github.issues.addLabels(context.repo({ number: issueNum, labels: [AUTO_CLOSE_LABEL] })),
+        context.github.issues.createComment(context.repo({ number: issueNum, body: commentText }))
+    ]);
+}
+
+/**
+ * Gets all issues on the current repository that match a search query
+ * @param {probot.Context} context Probot context for the current repository
+ * @param {string} searchQuery A search query to send to the GitHub API
+ * @returns {Promise<issue>} A list of issues that match the query
+ */
+async function queryIssues(context, searchQuery) {
+    return context.github.paginate(
+        context.github.search.issues({ q: searchQuery, per_page: 100 }),
+        result => result.data.items
+    );
+}
+
+/**
+ * Autocloses issues on a repository based on labels and activity level.
+ * @param {probot.Context} context Probot context for the current repository
+ * @returns {Promise<void>} A Promise that fulfills when the search is complete
+ */
+async function closeInactiveIssues(context) {
+
+    // repos that want auto close must first create a label
+    if (!await hasAutoCloseLabel(context)) {
+        return;
+    }
+
+    const [acceptedIssues, unacceptedIssues] = await Promise.all([
+        queryIssues(context, createAutoCloseAcceptedSearchQuery(context)),
+        queryIssues(context, createAutoCloseUnacceptedSearchQuery(context))
+    ]);
+
+    await Promise.all(acceptedIssues.map(
```

Comment:
Is there ever a chance of `acceptedIssues` and `unacceptedIssues` overlapping (it doesn't seem like it)? If not, thoughts on running both the `acceptedIssues` and `unacceptedIssues` operations in parallel rather than in sequence?

Reply:
No, because the accepted issues query includes `label:accepted` and the unaccepted issues query includes `-label:accepted`.

I tried doing it in parallel at first, but the code ended up really messy so I changed it back to this. We're not overly concerned about performance for this operation, so I think it's fine to do in sequence.

Reply:
Sounds good to me :+1: